src/features/surveys/components/SurveyEditor/blocks/OpenQuestionBlock.tsx

```diff
@@ -1,13 +1,222 @@
 import { ZetkinTextQuestion } from 'utils/types/zetkin';
+import {
+  Box,
+  ClickAwayListener,
+  MenuItem,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { FC, useEffect, useRef, useState } from 'react';
 
 interface OpenQuestionBlockProps {
-  question: ZetkinTextQuestion;
+  element: ZetkinTextQuestion;
+  inEditMode: boolean;
+  onEditModeEnter: () => void;
+  onEditModeExit: (question: Omit<ZetkinTextQuestion, 'required'>) => void;
 }
 
-const OpenQuestionBlock: FC<OpenQuestionBlockProps> = ({ question }) => {
-  return <Box>{JSON.stringify(question)}</Box>;
+enum fieldType {
+  multiLine = 'multiLine',
+  singleLine = 'singleLine',
+}
+
+type elementWithoutRequired = Omit<ZetkinTextQuestion, 'required'>;
+
+const OpenQuestionBlock: FC<OpenQuestionBlockProps> = ({
+  element,
+  inEditMode,
+  onEditModeEnter,
+  onEditModeExit,
+}) => {
+  const intl = useIntl();
+
+  const [typeField, setTypeField] = useState(
+    element.response_config.multiline === true
+      ? fieldType.multiLine
+      : fieldType.singleLine
+  );
+
+  const [title, setTitle] = useState(element.question);
+  const [description, setDescription] = useState(element.description);
+
+  const [responseConfig, setResponseConfig] = useState(
+    element.response_config.multiline
+  );
+  const [focus, setFocus] = useState<
+    'title' | 'description' | 'responseConfig' | null
+  >(null);
+
+  const titleRef = useRef<HTMLInputElement>(null);
+  const descriptionRef = useRef<HTMLInputElement>(null);
+  const typeConfigRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (focus === 'title') {
+      const input = titleRef.current;
+      input?.focus();
+    } else if (focus === 'description') {
+      const input = descriptionRef.current;
+      input?.focus();
+    } else {
+      const input = typeConfigRef.current;
+      input?.focus();
+    }
+  }, [focus]);
+
+  const handleKeyDown = (evt: React.KeyboardEvent<HTMLDivElement>) => {
+    if (evt.key === 'Enter') {
+      const elementWithout: elementWithoutRequired = {
+        description: description,
+        question: title,
+        response_config: { multiline: responseConfig },
+        response_type: element.response_type,
+      };
+      onEditModeExit(elementWithout);
+      setFocus(null);
+    }
+  };
+
+  const handleSelect = (event: React.BaseSyntheticEvent) => {
+    setTypeField(event.target.value);
+  };
+
+  const createElementToUpdate = () => {
+    const elemenToUpdate: elementWithoutRequired = {
+      description: description,
+      question: title,
+      response_config: { multiline: responseConfig },
+      response_type: element.response_type,
+    };
+    return elemenToUpdate;
+  };
+
+  return (
+    <ClickAwayListener
+      onClickAway={() => {
+        const element = createElementToUpdate();
+        onEditModeExit(element);
+        setFocus(null);
+      }}
+    >
+      {!inEditMode ? (
+        <Box onClick={() => onEditModeEnter()}>
```

Comment:
I think it's a bit easier to read if the ternary is the othe way around - aka inEditMode ? (edit mode) : (preview mode)

Reply:
Totally! it has more sense!

Reply:
Please, for my sanity, don't use ternaries where the expressions are more than a single or few lines of code. In this case, the condition is on line 106, but (I think) the expression if false starts on line 153. To me that makes it very difficult to assess the flow as I read the code.